pallets/protos/src/lib.rs

```diff
@@ -848,55 +851,91 @@ pub mod pallet {
 				map.insert(hex::encode(proto_id), Value::Object(Map::new()));
 			}
 
-			if params.return_owners {
+			if params.return_owners || !params.metadata_keys.is_empty() {
 				for (proto_id, map_proto) in map.iter_mut() {
-					let array_proto_id: Hash256 =
-						hex::decode(proto_id).unwrap()[..].try_into().unwrap();
-
-					let owner = <Protos<T>>::get(array_proto_id).unwrap().owner;
-
-					let string_owner = match owner {
-						ProtoOwner::User(account_id) => hex::encode(account_id.as_ref()), //format!("{:?}", account_id),
-						ProtoOwner::ExternalAsset(_linked_asset) => String::from("ExternalAsset"),
+					let array_proto_id: Hash256 = if let Ok(array_proto_id) = hex::decode(proto_id)
+					{
+						if let Ok(array_proto_id) = array_proto_id.try_into() {
+							array_proto_id
+						} else {
+							continue;
```

Comment:
I think it makes more sense to return an error, rather than ignoring it with `continute`. Cos if this line ever gets executed, it's means our code in `get_protos` screwed up somewhere

Reply:
we continue because rest can be still valid data, it should really not happen but we might wanna log it at least

Reply:
`proto_id` comes from user rpc request? hmm might wanna throw Error if so